dpdispatcher/submission.py

```diff
@@ -519,17 +519,18 @@ def handle_unexpected_job_state(self):
             raise RuntimeError("job_state for job {job} is unknown".format(job=self))
 
         if job_state == JobStatus.terminated:
-            dlog.info(f"job: {self.job_hash} {self.job_id} terminated; restarting job")
-            if self.fail_count > 3:
-                raise RuntimeError("job:job {job} failed 3 times".format(job=self))
             self.fail_count += 1
```

Comment:
This behavior may not be correct when it is recovered from the error: +1, error, restart, +1, error, restart...

Reply:
Ok, Maybe raise an error and reset fail_count 0? or every N times failed raise an error?

Reply:
What is usually causing JobStatus to be terminated? Is it an error of hpc resource allocations or timeout?

Reply:
1. some system like dpcloudserver maybe unstable. some jobs may be killed while running.

2. some command  cannot be executed correctly due to environment. (like cannot find `gmx` or `lmp` command, )

Reply:
@felix5572 I assume in most cases it would be the second situation. It would cost us core time and debug effort if these incorrect tasks are reattempted many times. Is there any way to forward shell error to `dlog.info`?

Reply:
Agree,  I will have a think.

Reply:
In the following snippet:
https://github.com/deepmodeling/dpdispatcher/blob/87e3ee8311a1a6e4b172b4979321e45b015d23bf/dpdispatcher/pbs.py#L56-L60
I don't have a lot of experience in standard error outputs. Is there something in stderr which will be a flag of task **actually** being killed, or will stderr report incorrect environment? It seems we can detect it in `err_str`.

Reply:
We only use exit code to determine whether the program exits normally.